### DISCUSSION TITLE
Feature Request: Read-only disclosure of private balances ("selective disclosure")

### DISCUSSION THREAD

**Discussion by arthurgousset** [celo-org/social-connect#219] [Feature requests (Needs)]
Surfacing this need from a recent conversation I had: 

> I would like to be able to comply with any national regulations and laws and have a way to selectively disclose to the government or other parties read-only access to my transactions.

**Comment by avaziri** [comment]
Please see celo-org/celo-monorepo#9293 for some context on the need for private balances and private transactions.

Assuming those features exist, there are a few scenarios where one would need to be able to disclose their role in transactions:

- In real-time, for single transactions between a group of trusted people. For example, think of private transactions in Venmo. Both you and the receiving party know each other's identities and the amount transferred, but no one else is given any information. You have to know who paid you, to know that the debt has been settled.
- Retroactively, to reveal to a trusted party your role in an exhaustive list of transactions in a certain time period. I suppose some sort of cryptographic proof would need to be employed to show that no transactions had been omitted. Regulators are considering banning or restricting cryptocurrencies due to concerns around money laundering, tax evasion, or other financial crimes. Rather than see privacy as in conflict with legal compliance, we can find a technical solution. By default, no one needs to make anything public but if they are compelled they can give a read-only proof that covers a limited time period, preserving their privacy as much as possible. This is much better than being compelled to give up your private key.

**Comment by arthurgousset** [comment]
Re-posting a [response](https://github.com/celo-org/celo-monorepo/discussions/9293#discussioncomment-2585212) I shared in celo-org/celo-monorepo#9293 for visibility (because this feature request is similar):
 
> Brief follow-up for visibility: @albertomartin shared a post about the technical priorities for cLabs:
> 
> * [What’s next for Celo. A technical roadmap, with input from sessions at Celo Connect](https://forum.celo.org/t/whats-next-for-celo-a-technical-roadmap-with-input-from-sessions-at-celo-connect/3313)
> 
> It includes a sentence that might be relevant here:
> 
> > Private transactions and private balances are areas that we want to explore more, especially with compliance in mind. We would love to get a pulse from you all on interest here.
> 
> I linked the following two feature requests in the comment section (for visibility):
> 
> * [Feature Request: Balance privacy for accounts with a phone number mapping #9293](https://github.com/celo-org/celo-monorepo/discussions/9293)
> * [Feature Request: Read-only disclosure of private balances ("selective disclosure") #9294](https://github.com/celo-org/celo-monorepo/discussions/9294)
> 
> Feel free to comment too @avaziri!

**Comment by arthurgousset** [comment]
When transferring this discussion topic:
- from: `celo-org/celo-monorepo/discussions`
- to: `celo-org/identity/discussions`
the upvotes were reset.

I'm including a screenshot of the upvotes for future reference: 

<img width="600" alt="image" src="https://user-images.githubusercontent.com/46296830/167644672-3a2f92bc-1b14-4633-a015-91b5f132e314.png">
